modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -74,21 +77,28 @@ const EditorUpload = function (editor: Editor): EditorUpload {
     return content;
   };
 
-  const replaceImageUrl = function (content: string, targetUrl: string, replacementUrl: string): string {
-    content = replaceString(content, 'src="' + targetUrl + '"', 'src="' + replacementUrl + '"');
+  const replaceImageUrl = (content: string, targetUrl: string, replacementUrl: string, usePlaceholder?: boolean): string => {
+    if (usePlaceholder) {
+      const placeholder = Settings.getPlaceholder(editor);
+
+      content = replaceString(content, 'src="' + targetUrl + '"', 'src="' + replacementUrl + '" data-mce-placeholder="' + placeholder + '" ');
+    } else {
+      content = replaceString(content, 'src="' + targetUrl + '"', 'src="' + replacementUrl + '"');
+    }
+
     content = replaceString(content, 'data-mce-src="' + targetUrl + '"', 'data-mce-src="' + replacementUrl + '"');
 
     return content;
   };
 
-  const replaceUrlInUndoStack = function (targetUrl: string, replacementUrl: string) {
-    Arr.each(editor.undoManager.data, function (level) {
+  const replaceUrlInUndoStack = (targetUrl: string, replacementUrl: string, replace: ReplaceFunction = replaceImageUrl) => {
```

Comment:
Why are we passing through a replace function here when it's always the same function, just with a different flag? It'd be cleaner and less code to just passthrough the flag instead.

Alternatively, you could just fork on the url. eg `replacementUrl === Env. transparentSrc`

Reply:
Given this looks like it was only used for the incorrect usage of the placeholder string, it should just be able to be reverted I believe.

Reply:
Reverting this would just revert all the changes. What should be done instead?

Reply:
You'll likely need to manually revert it then. You can compare changes to the develop branch, but if you're still stuck DM me on slack and we'll go through it more.

Reply:
No, I mean, if this is all removed then the rest of the PR goes as well, it's all kinda hinging on this function.

Reply:
It's not... you don't need to pass the `replace` argument through here, so that's all you need to remove/revert. The reason don't need the `replace` argument is you don't need the `usePlaceholder` argument (as the reason that was added is incorrect as noted in my other comment) and that's the only reason you need to pass a custom replace function.